webapp/graphite/storage.py

```diff
@@ -74,8 +76,9 @@ def find(self, pattern, startTime=None, endTime=None, local=False):
         if node.is_leaf:
           leaf_nodes.append(node)
         elif node.path not in found_branch_nodes: #TODO need to filter branch nodes based on requested interval... how?!?!?
-          yield node
-          found_branch_nodes.add(node.path)
+          if nodeFilter(user, node.path):
```

Comment:
`nodeFilter` probably needs to be applied one level up:

```python
for node in nodes:
  if not nodeFilter(user, node.path):
      continue
  if node.is_leaf:
    …
```

This avoids changing the logic below: only allows nodes end up in `leaf_nodes` and there is no need for the `minimal_node_set_filtered` calculation L137-146.